subsys/settings/src/settings_zms.c

```diff
@@ -252,6 +255,46 @@ static int settings_zms_load_subtree(struct settings_store *cs, const struct set
 }
 #endif /* CONFIG_SETTINGS_ZMS_LOAD_SUBTREE_PATH */
 
+static ssize_t settings_zms_load_one(struct settings_store *cs, const char *name, char *buf,
+				     size_t buf_len)
+{
+	struct settings_zms *cf = CONTAINER_OF(cs, struct settings_zms, cf_store);
+	char r_name[SETTINGS_MAX_NAME_LEN + SETTINGS_EXTRA_LEN + 1];
+	ssize_t rc = 0;
+	uint32_t name_hash;
+
+	/* verify that name is not NULL */
+	if (!name || !buf) {
+		return -EINVAL;
+	}
+
+	name_hash = sys_hash32(name, strlen(name)) & ZMS_HASH_MASK;
+	for (int i = 0; i <= cf->hash_collision_num; i++) {
+		name_hash = ZMS_UPDATE_COLLISION_NUM(name_hash, i);
+		/* Get the name entry from ZMS */
+		rc = zms_read(&cf->cf_zms, ZMS_NAME_ID_FROM_HASH(name_hash), r_name,
+			      sizeof(r_name) - 1);
+		if (rc <= 0) {
+			/* Name doesn't exist */
+			continue;
+		}
+		/* Found a name, this might not include a trailing \0 */
+		r_name[rc] = '\0';
+		if (strcmp(name, r_name)) {
+			/* Names are not equal let's continue to the next collision hash
+			 * if it exists.
+			 */
+			continue;
+		}
+
+		/* At this steps the names are equal, let's read the data */
+		return zms_read(&cf->cf_zms, ZMS_NAME_ID_FROM_HASH(name_hash) + ZMS_DATA_ID_OFFSET,
```

Comment:
This doesn't really return the actual stored length as promised in the API description but `MIN(buffer_len, value_len)`, right?